tests/benchdnn/dnnl_common.cpp

```diff
@@ -366,8 +366,40 @@ int execute_and_wait(perf_function_t &exec_func, const dnnl_engine_t &engine,
 
     execute_unmap_args(args, dnnl_args);
 
-    auto status = exec_func(stream, dnnl_args);
-    DNN_SAFE(dnnl_stream_wait(stream), CRIT);
+    dnnl_status_t status = dnnl_runtime_error;
+    bool run_regular_exec = true;
+#if DNNL_GPU_RUNTIME == DNNL_RUNTIME_DPCPP
+    while (use_sycl_graph && is_gpu(engine)) {
```

Comment:
Why is `while` with a `break`s preferred here instead of just `if`s or a lambda?

Reply:
If the functionality can't be supported, the fallback to a regular execution must happen. Without `while` it's gonna be a more tangled connection between two or three `if`s. Replacing this code with lambda would be identical to this `while` as we need objects from checking part and executing part, can't break it down into two without passing those objects around which doesn't help with readability. So to day, it's just a style preference.